src/components/shared/MarkerContextMenu.js

```diff
@@ -232,12 +281,36 @@ class MarkerContextMenuImpl extends PureComponent<Props> {
         onShow={this._onShow}
         onHide={this._onHide}
       >
-        <MenuItem onClick={this.setStartRange}>
-          Set selection start time here
-        </MenuItem>
-        <MenuItem onClick={this.setEndRange}>
-          Set selection end time here
-        </MenuItem>
+        {this._isZeroDurationMarker(marker) ? (
+          <>
+            <MenuItem onClick={this.setStartRangeFromMarkerStart}>
+              Set selection start time here
+            </MenuItem>
+            <MenuItem onClick={this.setEndRangeFromMarkerEnd}>
+              Set selection end time here
+            </MenuItem>
+          </>
+        ) : (
+          <>
+            <SubMenu title="Set selection start ">
+              <MenuItem onClick={this.setStartRangeFromMarkerStart}>
+                From the start of this marker
+              </MenuItem>
+              <MenuItem onClick={this.setStartRangeFromMarkerEnd}>
```

Comment:
Also there's a case we need to guard against: it's now easier to set the "end" before the "start". So I think we should guard against that. I gave some thought on it, and now I think the best UX is that we should disable the options when they'd do that.

We can do that this way:
1. add some code at the start of `render` to compute `selectionStart` and `selectionEnd`.
```js
   const selectionStart = previewSelection.hasSelection
      ? previewSelection.selectionStart
      : committedRange.start;
  ... same for selectionEnd
  const markerStart = marker.start;
  const markerEnd = marker.end || marker.start;
```
2. Then in the MenuItems we can use this with a `disabled` attribute, for example:
```js
               <MenuItem onClick={this.setStartRangeFromMarkerEnd} disabled={markerEnd > selectionEnd}>
```
And similarly for `setStartRangeFromMarkerEnd`.

I'm not convinced yet that this is perfect but at least this will be better, we can adjust in a later patch.